drivers/serial/uart_stm32.c

```diff
@@ -621,16 +626,663 @@ static void uart_stm32_irq_callback_set(const struct device *dev,
+{
+	struct uart_dma_stream *tx_stream = CONTAINER_OF(work,
+			struct uart_dma_stream, timeout_work);
+
+	struct uart_stm32_data *data = CONTAINER_OF(tx_stream,
+			struct uart_stm32_data, dma_tx);
+
+	const struct device *dev = data->uart_dev;
+
+	uart_stm32_async_tx_abort(dev);
+
+	LOG_DBG("tx: async timeout");
+}
+
+static int uart_stm32_async_rx_buf_rsp(const struct device *dev, uint8_t *buf,
+				       size_t len)
+{
+	struct uart_stm32_data *data = DEV_DATA(dev);
+
+	LOG_DBG("replace buffer (%d)", len);
+	data->rx_next_buffer = buf;
+	data->rx_next_buffer_len = len;
+
+	return 0;
+}
+
+static int uart_stm32_async_init(const struct device *dev)
+{
+	struct uart_stm32_data *data = DEV_DATA(dev);
+	USART_TypeDef *UartInstance = UART_STRUCT(dev);
+
+	data->uart_dev = dev;
+
+	if (data->dma_rx.dma_name != NULL) {
+		data->dev_dma_rx = device_get_binding(data->dma_rx.dma_name);
+		if (data->dev_dma_rx == NULL) {
+			LOG_ERR("%s device not found", data->dma_rx.dma_name);
+			return -ENODEV;
+		}
+	}
+
+	if (data->dma_tx.dma_name != NULL) {
+		data->dev_dma_tx = device_get_binding(data->dma_tx.dma_name);
+		if (data->dev_dma_tx == NULL) {
+			LOG_ERR("%s device not found", data->dma_tx.dma_name);
+			return -ENODEV;
+		}
+	}
+
+	/* Disable both TX and RX DMA requests */
+	uart_stm32_dma_rx_disable(dev);
+	uart_stm32_dma_tx_disable(dev);
+
+	k_delayed_work_init(&data->dma_rx.timeout_work,
+			    uart_stm32_async_rx_timeout);
+	k_delayed_work_init(&data->dma_tx.timeout_work,
+			    uart_stm32_async_tx_timeout);
+
+	/* Configure dma rx config */
+	memset(&data->dma_rx.blk_cfg, 0, sizeof(data->dma_rx.blk_cfg));
+#if defined(STM32F1) || defined(STM32F2) ||\
+	defined(STM32F4) || defined(STM32L1)
+	data->dma_rx.blk_cfg.source_address =
+				LL_USART_DMA_GetRegAddr(UartInstance);
+#else
+	data->dma_rx.blk_cfg.source_address =
+				LL_USART_DMA_GetRegAddr(UartInstance,
+						LL_USART_DMA_REG_DATA_RECEIVE);
+#endif
+
+	data->dma_rx.blk_cfg.dest_address = 0; /* dest not ready */
+
+	if (data->dma_rx.src_addr_increment) {
+		data->dma_rx.blk_cfg.source_addr_adj = DMA_ADDR_ADJ_INCREMENT;
+	} else {
+		data->dma_rx.blk_cfg.source_addr_adj = DMA_ADDR_ADJ_NO_CHANGE;
+	}
+
+	if (data->dma_rx.dst_addr_increment) {
+		data->dma_rx.blk_cfg.dest_addr_adj = DMA_ADDR_ADJ_INCREMENT;
+	} else {
+		data->dma_rx.blk_cfg.dest_addr_adj = DMA_ADDR_ADJ_NO_CHANGE;
+	}
+
+	/* RX disable circular buffer */
+	data->dma_rx.blk_cfg.source_reload_en  = 0;
+	data->dma_rx.blk_cfg.dest_reload_en = 0;
+	data->dma_rx.blk_cfg.fifo_mode_control = data->dma_rx.fifo_threshold;
+
+	data->dma_rx.dma_cfg.head_block = &data->dma_rx.blk_cfg;
+	data->dma_rx.dma_cfg.user_data = (void *)dev;
+	data->rx_next_buffer = NULL;
+	data->rx_next_buffer_len = 0;
+
+	/* Configure dma tx config */
+	memset(&data->dma_tx.blk_cfg, 0, sizeof(data->dma_tx.blk_cfg));
+
+#if defined(STM32F1) || defined(STM32F2) ||\
+	defined(STM32F4) || defined(STM32L1)
```

Comment:
Please use Zephyr flags (`CONFIG_SOC_SERIES_STM32F1X`, ...)

Reply:
This request still stands

Reply:
Sorry! I fixed in one place and not in the other.